Group Finder

conrad-vanl opened the pull request:
MVP Implementation of the group finder.

Fixes #169 
Fixes #170 
Fixes #172 
Fixes #173 

## Creator

- [ ] Build relevant tests if any apply
- [x] Ensure there are no new warnings
- [ ] Upload an animated GIF showcasing the solution (if it applies)
- [x] Set a relevant reviewer

## Reviewer

- [ ] Run `test` and make sure all tests pass
- [ ] Review function and form on iOS
- [ ] Review function and form on Android
- [ ] Review function and form on Web
- [ ] Review new test logic

Lepozepo commented on src/@ui/NativeWebRouter/Link.js:
```diff
@@ -45,20 +45,23 @@ export default class Link extends Component {
 
```
Not too sure how this happened. But when clicking on a link web errors with this:

<img width="1088" alt="screen shot 2018-01-03 at 10 01 17 am" src="https://user-images.githubusercontent.com/2302110/34533269-57491ffc-f06e-11e7-9dd2-136f2c661a41.png">

It happens specifically when you click while waiting for group results to load. There may be something undefined while that's happening that should have a default?

conrad-vanl replied:
Fixed in 554796f

conrad-vanl commented on src/@data/mediaPlayer/albumsQuery.js:
```diff
@@ -4,7 +4,7 @@ export default gql`
   query getAlbums($limit: Int!, $skip: Int!) {
     library: content(channel: "newspring_albums", limit: $limit, skip: $skip) {
       id
-      entryId: id
```
Done in 5597ee9

I think that's right, as Apollo caches by `id` by default.

conrad-vanl commented on src/@ui/FeedView/FeedList.web.js:
```diff
@@ -50,14 +60,14 @@ class FeedList extends PureComponent {
       this.props.onEndReached();
```
Should be fixed in e151269f2f662039b295374da6e6f01bcfb260b8

conrad-vanl commented:
@Lepozepo I believe I addressed the issues you pointed about above. Before we can merge:

- [ ] we need to fix the infinite scrolling issues
- [ ] I'd like @burkeshartsis's opinion of `@ui/Card`
- [ ] If opinion above is positive, I'll add tests for `Card` and refactor `FeedItemCard` to use the same components

Lepozepo commented on src/@ui/forms/GroupSearchForm/index.js:
```diff
@@ -0,0 +1,128 @@
+    touched: PropTypes.shape({}),
+    errors: PropTypes.shape({}),
+    values: PropTypes.shape({}),
+    campuses: PropTypes.arrayOf(PropTypes.shape({})),
```
We should set these shapes!

Lepozepo commented:
@conrad-vanl I patched the graphql error, it should be ok (but brittle) now. Data patches continue [here](https://github.com/NewSpring/Apollos/issues/212)
